sdks/python/apache_beam/ml/inference/base.py

```diff
@@ -150,27 +155,24 @@ def update(
 
 class _RunInferenceDoFn(beam.DoFn):
   """A DoFn implementation generic to frameworks."""
-  def __init__(self, model_loader: ModelLoader, clock=None):
+  def __init__(self, model_loader: ModelLoader, clock):
     self._model_loader = model_loader
     self._inference_runner = model_loader.get_inference_runner()
     self._shared_model_handle = shared.Shared()
     self._metrics_collector = _MetricsCollector(
         self._inference_runner.get_metrics_namespace())
     self._clock = clock
-    if not clock:
-      self._clock = _ClockFactory.make_clock()
     self._model = None
 
   def _load_model(self):
     def load():
       """Function for constructing shared LoadedModel."""
       memory_before = _get_current_process_memory_in_bytes()
-      start_time = self._clock.get_current_time_in_microseconds()
+      start_time = _to_milliseconds(self._clock.time())
       model = self._model_loader.load_model()
-      end_time = self._clock.get_current_time_in_microseconds()
+      end_time = _to_milliseconds(self._clock.time())
```

Comment:
A couple questions here:
- why change the units to milliseconds?
- maybe we should use [time_ns](https://docs.python.org/3/library/time.html#time.time_ns) instead? The time docs say time_ns should be preferred to avoid precision loss. It's py 3.7+ only, but we've dropped support for 3.6 now

Reply:
If you look at the line I removed 172, the previous behavior was to get start and end time in microseconds and then convert to milliseconds down below.

For some reason the tensorflow team felt (and it seems right) that milliseconds was adequate for model loading, whereas microseconds were used for prediction measurement.

I think it's possible model loading can take hundreds or thousands of milliseconds whereas a prediction might not take a single millisecond.

Reply:
I see, you might consider using time.time_ns then to avoid floating point precision issues when measuring a single millisecond delta.

Reply:
This was new to version 3.7.  I think since we don't support 3.6 anymore we can do this now.  Done.